### DISCUSSION TITLE
Add Proxy settings

### DISCUSSION THREAD

**Discussion by treejadey** [Chatterino/chatterino2#3856] [Ideas]
## Idea:
It would be cool to get Proxy settings into Chatterino, because that would allow people to use Chatterino where Twitch is blocked (China and Iran), without the need of running the entire network through a VPN/Proxy.

## Why:
Since Chatterino has basically became an instant messaging system for certain people, this would absolutely improve the QOL of people using Chatterino in countries where Twitch servers are blocked.

## Why not:
1. It could take a lot of time and effort to implement, will result in QOL improvements for a small fraction of the userbase.
2. Additional settings in the already cluttered settings menu.

**Comment by pajlada** [comment]
I think this is a feature that has a place in Chatterino.

Reason 1 and 2 are bad reasons to reject this feature, however an owner of the feature would be good (preferably someone who would also want to use the feature, since that's not me).

https://doc.qt.io/qt-6/qnetworkproxy.html is a good place to start, note that some of our connections are not made through QNetworks so we'd need to look deeper into that for a solution.

**Comment by Mm2PL** [comment]
xref https://github.com/Chatterino/chatterino2/issues/450
